### ISSUE TITLE
Legacy bundle served when using Headless Chrome

### ISSUE THREAD

**Issue by rereradu** [meteor/meteor#11121]
I am using Meteor 1.10.2 on a Debian 10 Buster server. When loading the site in Headless Chrome (using `puppeteer`), the `web.legacy` bundle is used, which is causing lots of errors - my app doesn't support legacy browsers.

Example of error captured in the console: 

`no such file or directory, open '<PATH_TO_APPLICATION>/bundle/programs/web.browser.legacy/dynamic/node_modules/@babel/runtime/helpers/toConsumableArray.js'`

The user agent is set to this: 
`Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/83.0.4103.116 Safari/537.36`

Is there any other feature detection that determines which bundle to use? What's causing the app to serve the legacy bundle in this situation? 

**EDIT**

I am facing the same issue when using the `prerender` package (which is also using Headless Chrome). All requests to the app "go through" to the `web.browser.legacy` bundle. 

**EDIT 2**
Running `puppeteer` on a serverless Google Cloud Function has the same result. The bundle served is `web.browser.legacy`.

**Comment by hexsprite** [comment]
why don't you set the user agent in your puppeter instance ?

**Comment by rereradu** [comment]
Closing this issue, as I found the culprit (tldr: not a Meteor issue). Here's the explanation:
I was trying to create a pre-rendered version of the page to serve to crawlers / search engines but I failed to prevent client re-hydration and that's what was actually causing the errors.  The solution was to first test if the app has already been pre-rendered server side before calling the client-side code (in my case just checking if the document body had a specific element as its child).

And, of course, one of the things I tried before realising the problem was to set the user agent in puppeteer.

**Comment by rereradu** [comment]
Closing this issue, as I found the culprit (tldr: not a Meteor issue). Here's the explanation:
I was trying to create a pre-rendered version of the page to serve to crawlers / search engines but I failed to prevent client re-hydration and that's what was actually causing the errors. The solution was to first test if the app has already been pre-rendered server side before calling the client-side code (in my case just checking if the document body had a specific element as its child).

And, of course, one of the things I tried before realising the problem was to set the user agent in puppeteer.
